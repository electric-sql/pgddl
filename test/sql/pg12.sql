\pset null _null_

SET client_min_messages = warning;
SET ROLE postgres;

create table tab_generated12 (
       a integer generated by default as identity,
       b integer generated always as identity,
       c integer generated always as (a+b) stored,
       d integer default 42,
       e serial,
       f integer
);

-- select * from ddlx_describe('tab_generated12'::regclass) d;

\pset format unaligned
select ddlx_script('tab_generated12');
