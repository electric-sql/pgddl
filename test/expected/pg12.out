\pset null _null_
SET client_min_messages = warning;
SET ROLE postgres;
create table tab_generated12 (
       a integer generated by default as identity,
       b integer generated always as identity,
       c integer generated always as (a+b) stored,
       d integer default 42,
       e serial,
       f integer
);
-- select * from ddlx_describe('tab_generated12'::regclass) d;
\pset format unaligned
select ddlx_script('tab_generated12');
ddlx_script
BEGIN;

/*
ALTER TABLE tab_generated12 ALTER e DROP DEFAULT;

DROP TABLE tab_generated12;
*/

-- Type: TABLE ; Name: tab_generated12; Owner: postgres

CREATE TABLE tab_generated12 (
    a integer GENERATED BY DEFAULT AS IDENTITY,
    b integer GENERATED ALWAYS AS IDENTITY,
    c integer GENERATED ALWAYS AS (a + b) STORED,
    d integer,
    e integer NOT NULL,
    f integer
);

COMMENT ON TABLE tab_generated12 IS NULL;

ALTER TABLE tab_generated12 ALTER d SET DEFAULT 42;

ALTER SEQUENCE tab_generated12_e_seq OWNED BY tab_generated12.e;


-- DEPENDANTS

ALTER TABLE tab_generated12 ALTER e SET DEFAULT nextval('tab_generated12_e_seq'::regclass);

END;

(1 row)
